test/arrays.coffee

```diff
@@ -70,6 +68,32 @@ test "array splats with nested arrays", ->
   list = [1, 2, a...]
   arrayEq list, [1, 2, [nonce]]
 
+test "#4260: splat after existential operator soak", ->
+  a = {b: [3]}
+  arrayEq [a?.b...], [3]
+  arrayEq [c?.b...], []
+  arrayEq [...a?.b], [3]
+  arrayEq [...c?.b], []
+
+  # Should not trigger implicit call, e.g. rest ... => rest(...)
+  arrayEq [... a?.b], [3]
+  arrayEq [... c?.b], []
+  arrayEq [a?.b ...], [3]
+  arrayEq [c?.b ...], []
+
+test "#1349: trailing if after splat", ->
+  a = [3]
+  b = yes
+  c = null
+  arrayEq [a if b...], [3]
+  arrayEq [a if c...], []
+  arrayEq [...a if b], [3]
+  arrayEq [...a if c], []
+
+  # Should not trigger implicit call, e.g. rest ... => rest(...)
+  arrayEq [a if b ...], [3]
+  arrayEq [a if c ...], []
```

Comment:
Absolutely last, final thing: this could also do with a spaced prefixed version.